telemetry/prometheus/metrics.go

```diff
@@ -24,6 +24,7 @@ type metrics struct {
 
 // NewMetrics initializes a new instance of Prometheus metrics.
 func NewMetrics(cfg *Config) (*metrics, error) { //nolint:revive // only used as Metrics interface.
+	setDefaultCfg(cfg)
```

Comment:
I think the 2 config are TOTALLY optional, since some metrics instance does not even emit `Time` or `Histogram` stats - so they should not bother to set those 2 values.

Thus choose to provide a default value here.